lib/functions/Core.ts

```diff
@@ -45,6 +89,27 @@ export abstract class BaseFunction<Operator> {
 
   protected abstract handleInvalidTypes(args: Term[]): never;
 
+  /**
+   * Searches in a depth fist way for the best matching overload.
+   * @param args
+   * @private
+   */
+  private search(args: Term[]): E.SimpleApplication {
```

Comment:
Does this needs checks in case `getImplementation` is `undefined`?

Reply:
No, this should also return `| undefined` my bad.

Reply:
We should urgently enable strict compilation in tsconfig, to avoid issues like this.